controllers/resource/fetcher.go

```diff
@@ -589,20 +589,22 @@ func MapMachineTemplateToVSphereMachineConfigSpec(vsMachineTemplate *vspherev1.V
 
 func MapClusterToCloudStackDatacenterConfigSpec(csCluster *cloudstackv1.CloudStackCluster) *anywherev1.CloudStackDatacenterConfig {
 	csSpec := &anywherev1.CloudStackDatacenterConfig{}
-	var zones []anywherev1.CloudStackZone
-	for _, csZone := range csCluster.Spec.Zones {
-		zones = append(zones, anywherev1.CloudStackZone{
-			Id:   csZone.ID,
-			Name: csZone.Name,
-			Network: anywherev1.CloudStackResourceIdentifier{
-				Id:   csZone.Network.ID,
-				Name: csZone.Network.Name,
+	var azs []anywherev1.CloudStackAvailabilityZone
+	for _, fd := range csCluster.Spec.FailureDomains {
```

Comment:
I think we've answered this before - but will there be any point at which this updated eks-a controller is reading in v1beta1 CAPC resources, and has to map them back to AZ's? This code assumes we always have FailureDomains present